pallets/author-mapping/src/lib.rs

```diff
@@ -72,6 +73,10 @@ pub mod pallet {
 		type DepositCurrency: Currency<Self::AccountId> + ReservableCurrency<Self::AccountId>;
 		/// The amount that should be taken as a security deposit when registering a NimbusId.
 		type DepositAmount: Get<<Self::DepositCurrency as Currency<Self::AccountId>>::Balance>;
+		/// Additional keys
+		/// Convertible From<NimbusId> to get default keys for each mapping (for the migration)
+		/// TODO: add lookup trait bound to find key in keys, see sp_runtime::traits::OpaqueKeys
+		type Keys: Parameter + Member + MaybeSerializeDeserialize + From<NimbusId>;
```

Comment:
I think you can require the From<NimbusId> in the migration struct if that's what you need.

Reply:
This is not only required for the migration, it's also require for the new behavior of old extrinsics that we keep for backward compatibility